File: app/src/modules/payment/dtos/payment-provider-test-data-response-dto.ts
import {PaymentRequest} from '../../../models/payment/payment-request';
import {PaymentInformation} from '../../../models/payment/payment-information';

export interface PaymentProviderTestDataResponseDTO {
    ok: boolean;
    request?: PaymentRequest | null;
    paymentInformation?: PaymentInformation | null;
    errorMessage?: string | null;
}

File: backend/src/main/java/de/aivot/prosuna/backend/payment/dtos/PaymentProviderTestDataResponseDTO.java
package de.aivot.prosuna.backend.payment.dtos;

import de.aivot.prosuna.backend.payment.models.XBezahldienstePaymentRequest;
import de.aivot.prosuna.backend.payment.models.XBezahldienstePaymentTransaction;

import jakarta.annotation.Nonnull;
import jakarta.annotation.Nullable;

public record PaymentProviderTestDataResponseDTO(
        @Nonnull Boolean ok,
        @Nullable XBezahldienstePaymentRequest request,
        @Nullable XBezahldienstePaymentTransaction transaction,
        @Nullable String errorMessage
) {
}
